tests/std/tests/P0896R4_ranges_iterator_machinery/test.cpp

```diff
@@ -3370,6 +3370,35 @@ namespace vso1121031 {
     STATIC_ASSERT(!has_member_value_type<indirectly_readable_traits<iterish<float const volatile>>>);
 } // namespace vso1121031
 
+namespace lwg3541 {
+    // Make indirectly_readable_traits SFINAE friendly
+    using std::indirectly_readable_traits, std::same_as;
+
+    template <class Element>
+    struct iterish {
+        using value_type   = int;
+        using element_type = Element;
+    };
+
+    template <class Element, class Enable = void>
+    struct SFINAE {};
+
+    // Specialization only available for T which respect the traits.
+    template <class Element>
+    struct SFINAE<Element, std::enable_if_t<has_member_value_type<indirectly_readable_traits<iterish<Element>>>>> {
+        using value_type = typename indirectly_readable_traits<iterish<Element>>::value_type;
+    };
+    STATIC_ASSERT(same_as<SFINAE<int>::value_type, int>);
+    STATIC_ASSERT(same_as<SFINAE<int const>::value_type, int>);
+    STATIC_ASSERT(same_as<SFINAE<int volatile>::value_type, int>);
+    STATIC_ASSERT(same_as<SFINAE<int const volatile>::value_type, int>);

```

Comment:
Why not simply `STATIC_ASSERT(same_as<indirectly_readable_traits<iterish<int meow>>, int>)`? And don't we already have coverage for these cases elsewhere?

Reply:
Wasnt the whole point of the issue to ensure that it is SFINAE friendly?

AFAIK `STATIC_ASSERT(same_as<indirectly_readable_traits<iterish<int meow>>, int>)` does not check that?

Reply:
We can't determine if something is SFINAE-friendly with test cases that don't cause substitution failure. The `iterish<float meow>` test cases validate the point of the change; without it, `has_member_value_type<indirectly_readable_traits<iterish<float meow>>>`  would produce a compile error instead of evaluating to `false`. _These_ test cases are simply verifying that things that worked before continue to work, which - as I suggested - I suspect we have coverage of elsewhere.

Reply:
Interestingly we already have exactly that test directly above.

This suggests, that the change is not really needed. I have removed the duplicated tests

Reply:
> Interestingly we already have exactly that test directly above. This suggests, that the change is not really needed. I have removed the duplicated tests

Oh 🤦. This is the issue I've been talking with our compiler devs and EDG about for the last couple of weeks: MSVC and Clang non-conformingly are treating ambiguity while choosing a partial template specialization as being a SFINAE-able error in cases where they should not be doing so. The test cases were passing because of the bug, but will need the product code change to keep working once the compiler bugs are fixed.

Reply:
I'm leaving this unresolved - despite that there's nothing here to resolve - to point out why we're adding no new test coverage in this PR.